pandas/io/formats/format.py

```diff
@@ -565,71 +583,208 @@ def __init__(
+    def columns(self) -> Index:
+        return self._columns
+
+    @columns.setter
+    def columns(self, columns: Optional[Sequence[str]]) -> None:
+        if columns is not None:
+            self._columns = ensure_index(columns)
+            self.frame = self.frame[self._columns]
+        else:
+            self._columns = self.frame.columns
+        assert self._columns is not None
+
+    @property
+    def col_space(self) -> ColspaceType:
+        return self._col_space
+
+    @col_space.setter
+    def col_space(self, col_space: Optional[ColspaceArgType]) -> None:
+        self._col_space: ColspaceType
         if col_space is None:
-            self.col_space = {}
+            self._col_space = {}
         elif isinstance(col_space, (int, str)):
-            self.col_space = {"": col_space}
-            self.col_space.update({column: col_space for column in self.frame.columns})
+            self._col_space = {"": col_space}
+            self._col_space.update({column: col_space for column in self.frame.columns})
         elif isinstance(col_space, Mapping):
             for column in col_space.keys():
                 if column not in self.frame.columns and column != "":
                     raise ValueError(
                         f"Col_space is defined for an unknown column: {column}"
                     )
-            self.col_space = col_space
+            self._col_space = col_space
         else:
-            if len(frame.columns) != len(col_space):
+            if len(self.frame.columns) != len(col_space):
                 raise ValueError(
                     f"Col_space length({len(col_space)}) should match "
-                    f"DataFrame number of columns({len(frame.columns)})"
+                    f"DataFrame number of columns({len(self.frame.columns)})"
                 )
-            self.col_space = dict(zip(self.frame.columns, col_space))
+            self._col_space = dict(zip(self.frame.columns, col_space))
+        assert self._col_space is not None
 
-        self.header = header
-        self.index = index
-        self.line_width = line_width
-        self.max_rows = max_rows
-        self.min_rows = min_rows
-        self.max_cols = max_cols
-        self.max_rows_displayed = min(max_rows or len(self.frame), len(self.frame))
-        self.show_dimensions = show_dimensions
-        self.table_id = table_id
-        self.render_links = render_links
+    @property
+    def max_rows_displayed(self) -> int:
+        return min(self.max_rows or len(self.frame), len(self.frame))
 
-        if justify is None:
-            self.justify = get_option("display.colheader_justify")
+    def _is_in_terminal(self) -> bool:
+        """Check if the output is to be shown in terminal."""
+        return bool(self.max_cols == 0 or self.max_rows == 0)
+
+    @property
+    def max_cols_adj(self) -> Optional[int]:
+        """Number of columns fitting the screen."""
+        self._max_cols_adj: Optional[int]
+
+        try:
```

Comment:
is this new?

Reply:
i c was a hasattr check before (can you restore). I suspect once you change the class structure this won't be needed anyhow (or rather it will be used / needed in some of the classes but not all)

Reply:
@jreback, if you inquire about try/except construction, there there is a reason for it.
There is an assignment to max_cols_adj at line 974. After that assignment I want only that value to be used. Therefore, in the property I want to return ``self._max_cols_adj`` right away if it exists.
Technically, this can be refactored so that the setter takes care of ``self._max_cols_adj`` value.

Yes, this property will be needed only for console and html outputs.
I intended to make classes separation afterwards.
Do you still suggest that I restore the previous implementation? Because there was too much going on in the ``_chk_truncate`` method.

Reply:
Or do you suggest that I use hasattr here instead of try/except? I would probably better go for a setter for more clarity.